### PR TITLE
docs(ui_storage): add Firebase UI Storage docs

### PR DESCRIPTION
**Pull request by lesnitsky**
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] All unit tests pass (`melos run test:unit:all` doesn't fail).
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is _not_ a breaking change.

[issue database]: https://github.com/firebase/FirebaseUI-Flutter/issues
[Contributor Guide]: https://github.com/firebase/FirebaseUI-Flutter/blob/main/docs/contributing.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_ui_storage/doc/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+- [`KeepOriginalNameUploadPolicy`] – keeps the original file name
+- [`KeepPathUploadPolicy`] - keeps the original file path
+- [`UuidFileUploadNamingPolicy`] – generates a UUID for each uploaded file
+
+[`KeepOriginalNameUploadPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/KeepOriginalNameUploadPolicy-class.html
+[`KeepPathUploadPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/KeepPathUploadPolicy-class.html
+[`UuidFileUploadNamingPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/UuidFileUploadNamingPolicy-class.html
+
+If you ever need to override this top-level configuration, you can do so by using `FirebaseUIStorageConfigOverride`:
```
Comment (exaby73) [inline packages/firebase_ui_storage/doc/getting-started.md:92]: Could we possibly mention the default here?

File: packages/firebase_ui_storage/doc/paginated-loading-controller.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# PaginatedLoadingController
+
+When loading large amounts of objects from the Firebase Storage, it makes sense to load them in batches, but requires handling of the pagination logic.
+
+Firebase UI Storage provides [`StorageListView`] and [`StorageGridView`] widgets that implement pagination logic for you, and allow to load objects from the Firebase Storage in batches while scrolling.
+
+In cases where you want to build a completely custom scroll view, but still have pagination logic handled – you can use [`PaginatedLoadingController`].
+
+```dart
```
Comment (exaby73) [inline packages/firebase_ui_storage/doc/paginated-loading-controller.md:12]: Could we add examples of using `.load()` and `.shouldLoatNextPage()` as well?

**Review by exaby73 — CHANGES_REQUESTED** [review]
Couple of nits, else LGTM

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/firebase_ui_storage/README.md [hunk 3]
```diff
@@ -4,171 +4,27 @@
...
-Download Firebase project config
+- [`PaginatedLoadingController`](https://github.com/firebase/FirebaseUI-Flutter/tree/main/packages/firebase_ui_storage/doc/paginated-loading-controller.md)
 
-```sh
-flutterfire configure
-```
+## Issues
 
-## Configuration
+Please file Firebase UI Storage specific issues, bugs, or feature requests in our [issue tracker].
```
[the remark this answers is not in the document]
Comment (lesnitsky) [reply packages/firebase_ui_storage/README.md:22]: https://github.com/firebase/FirebaseUI-Flutter/blob/docs/ui-storage/packages/firebase_ui_storage/README.md?plain=1#L28

File: packages/firebase_ui_storage/doc/getting-started.md (hunk 1 above)
Comment (lesnitsky) [reply packages/firebase_ui_storage/doc/getting-started.md:92]: aa8ee60bf6fecc14d7b0606527f01c138433074b

File: packages/firebase_ui_storage/doc/paginated-loading-controller.md (hunk 2 above)
Comment (lesnitsky) [reply packages/firebase_ui_storage/doc/paginated-loading-controller.md:12]: 88b5cd26f4965ba9dce6900c5598ebfb20d610ca

File: packages/firebase_ui_storage/doc/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+## Installation
+
+Install dependencies
+
+```sh
+flutter pub add firebase_core firebase_storage firebase_ui_storage
+```
+
+Download Firebase project config
```
[the remark this answers is not in the document]
Comment (lesnitsky) [reply packages/firebase_ui_storage/doc/getting-started.md]: 5d808ff

File: packages/firebase_ui_storage/doc/upload-button.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          content: Text('Upload complete: ${ref.fullPath}'),
+        ),
+      ),
+      variant: ButtonVariant.filled,
+    );
+  }
+}
+```
+
+Upload buttons uses `uploadRoot` and `namingPolicy` from [`FirebaseUIStorageConfiguration`](https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/FirebaseUIStorageConfiguration-class.html) to determine the location and file name of the uploaded file.
```
[the remark this answers is not in the document]
Comment (lesnitsky) [reply packages/firebase_ui_storage/doc/upload-button.md:30]:
```suggestion
`UploadButton` uses `uploadRoot` and `namingPolicy` from [`FirebaseUIStorageConfiguration`](https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/FirebaseUIStorageConfiguration-class.html) to determine the location and file name of the uploaded file.
```

File: packages/firebase_ui_storage/doc/upload-button.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    );
+  }
+}
+```
+
+`UploadButton` uses `uploadRoot` and `namingPolicy` from [`FirebaseUIStorageConfiguration`](https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/FirebaseUIStorageConfiguration-class.html) to determine the location and file name of the uploaded file.
+
+## Showing upload progress
+
+`UploadButton` shows `CircularProgressIndicator` under `MaterialApp`, and `CupertinoActivityIndicator` under `CupertinoApp` while the upload is in progress. For large uploads, you might want to show upload progress, so you could replace the button with a [`TaskProgressIndicator`](<(https://github.com/firebase/FirebaseUI-Flutter/tree/main/packages/firebase_ui_storage/doc/task-progress-indicator.md)>) in a stateful widget using `onUploadStarted`:
```
Comment (lesnitsky) [inline packages/firebase_ui_storage/doc/upload-button.md:34]:
```suggestion
`UploadButton` shows `CircularProgressIndicator` under `MaterialApp`, and `CupertinoActivityIndicator` under `CupertinoApp` while the upload is in progress. For large uploads, you might want to show upload progress, so you could replace the button with a [`TaskProgressIndicator`](https://github.com/firebase/FirebaseUI-Flutter/tree/main/packages/firebase_ui_storage/doc/task-progress-indicator.md) in a stateful widget using `onUploadStarted`:
```

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**